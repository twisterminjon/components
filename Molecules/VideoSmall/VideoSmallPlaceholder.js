import React from 'react'
import PropTypes from 'prop-types'

import useUser from '../../../hooks/useUser'

import Avatar from '../../Atoms/Avatar/Avatar'

import './VideoSmall.css'

VideoSmallPlaceholder.propTypes = {
  /** User id of user to query */
  userId: PropTypes.string,
}

export default function VideoSmallPlaceholder({ userId = '' }) {
  const { profileImage, displayName, loading } = useUser(userId)

  return <VideoSmallPlaceholderView profileImage={profileImage} displayName={displayName} loading={loading} />
}

export function VideoSmallPlaceholderView({ profileImage, displayName, loading }) {
  return (
    <div className="videosmall-placeholder-wrap">
      <Avatar loading={loading} displayName={displayName} size="large" imgUrl={profileImage} />
    </div>
  )
}
