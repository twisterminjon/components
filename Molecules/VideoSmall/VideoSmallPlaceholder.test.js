import React from 'react'
import { VideoSmallPlaceholderView } from './VideoSmallPlaceholder'

describe('VideoSmallPlaceholder', () => {
  it('matches the base snapshot', () => {
    const wrapper = window.shallow(<VideoSmallPlaceholderView />)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot with loading', () => {
    const wrapper = window.shallow(<VideoSmallPlaceholderView loading={true} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot with profile image', () => {
    const wrapper = window.shallow(<VideoSmallPlaceholderView profileImage={'https://www.fillmurray.com/400/400'} />)
    expect(wrapper).toMatchSnapshot()
  })
})
