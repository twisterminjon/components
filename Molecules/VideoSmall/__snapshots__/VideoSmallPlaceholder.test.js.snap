// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoSmallPlaceholder matches the base snapshot 1`] = `
<div
  className="videosmall-placeholder-wrap"
>
  <Avatar
    className=""
    dimmed={false}
    displayName={null}
    imgUrl=""
    loading={false}
    size="large"
    style={Object {}}
  />
</div>
`;

exports[`VideoSmallPlaceholder matches the snapshot with loading 1`] = `
<div
  className="videosmall-placeholder-wrap"
>
  <Avatar
    className=""
    dimmed={false}
    displayName={null}
    imgUrl=""
    loading={true}
    size="large"
    style={Object {}}
  />
</div>
`;

exports[`VideoSmallPlaceholder matches the snapshot with profile image 1`] = `
<div
  className="videosmall-placeholder-wrap"
>
  <Avatar
    className=""
    dimmed={false}
    displayName={null}
    imgUrl="https://www.fillmurray.com/400/400"
    loading={false}
    size="large"
    style={Object {}}
  />
</div>
`;
